src/TheRestOfYourLife/hittable_list.h

```diff
@@ -63,19 +71,23 @@ bool hittable_list::bounding_box(double t0, double t1, aabb& output_box) const {
     return true;
 }
 
-bool hittable_list::hit(const ray& r, double t_min, double t_max, hit_record& rec) const {
-    hit_record temp_rec;
-    bool hit_anything = false;
-    double closest_so_far = t_max;
 
-    for (int i = 0; i < list_size; i++) {
-        if (list[i]->hit(r, t_min, closest_so_far, temp_rec)) {
-            hit_anything = true;
-            closest_so_far = temp_rec.t;
-            rec = temp_rec;
-        }
-    }
-    return hit_anything;
+double hittable_list::pdf_value(const vec3& o, const vec3& v) const {
+    auto weight = 1.0/objects.size();
+    auto sum = 0.0;
+
+    for (auto object : objects)
+        sum += weight * object->pdf_value(o, v);
+
+    return sum;
 }
 
+
+vec3 hittable_list::random(const vec3 &o) const {
+    auto int_size = static_cast<int>(objects.size());
+    auto index = static_cast<size_t>(random_int(0, int_size-1));
```

Comment:
why the cast back to `size_t`?

Reply:
Just being explicit about the conversion from signed to unsigned. But it looks like my compiler doesn't care either, so I'll remove the cast.